sbndaq-artdaq/Generators/ICARUS/icarus-base/PhysCrate.cc

```diff
@@ -59,23 +59,30 @@ PhysCrate::initialize(std::vector<int> busVec)
       
     }
     else{
+      int link = 0;
       for(int nBus : busVec){
 
-	do {        
-	  TRACEN("PhysCrate.cc", TLVL_INFO, "trying bus=%d,dev=%d",nBus,nDev);
+	for(unsigned int nBoardInCrate = 0; nBoardInCrate < boardsinCrateVec[link]; ++nBoardInCrate)
+	{        
+	  TRACEN("PhysCrate.cc", TLVL_INFO, "trying bus=%d,dev=%d",nBus,nBoardInCrate);
 	  
-	  boards[nBoards]=new A2795Board(nDev,nBus);
+	  boards[nBoards]=new A2795Board(nDev,nBoardInCrate);
```

Comment:
Shouldn't `nBoardInCrate` be replacing `nDev` instead of `nBus`?

```suggestion
	  boards[nBoards]=new A2795Board(nBoardInCrate,nBus);
```

Reply:
yes you are right. that should be fixed now